pymc_extras/statespace/models/structural/components/seasonality.py

```diff
@@ -332,14 +348,19 @@ def populate_component_properties(self):
                 "constraints": "Positive",
                 "dims": None,
             }
-            self.shock_names = [f"{self.name}[{name}]" for name in self.observed_state_names]
+            if self.share_states:
+                self.shock_names = [f"{self.name}[shared]"]
+            else:
+                self.shock_names = [f"{self.name}[{name}]" for name in self.observed_state_names]
```

Comment:
This is gonna conflict with `main` because of #559 . The following should fix the conflict:
```suggestion
if self.innovations:
    self.param_names += [f"sigma_{self.name}"]
    self.param_info[f"sigma_{self.name}"] = {
        "shape": () if k_endog_effective == 1 else (k_endog_effective,),
        "constraints": "Positive",
        "dims": None if k_endog_effective == 1 else (f"endog_{self.name}",),
    }
    if k_endog_effective > 1:
        self.param_dims[f"sigma_{self.name}"] = (f"endog_{self.name}",)

    if self.share_states:
        self.shock_names = [f"{self.name}[shared]"]
    else:
        self.shock_names = [f"{self.name}[{name}]" for name in self.observed_state_names]
```